relay-crash/src/lib.rs

```diff
@@ -13,19 +14,63 @@ mod native {
     include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
 }
 
+/// A transport function used to send envelopes to Sentry.
+pub type Transport = fn(envelope: &[u8]);
+
+/// Serializes a sentry_native envelope and passes the buffer to the [`Transport`] function.
+unsafe extern "C" fn transport_proxy(
+    envelope: *const native::sentry_envelope_s,
+    tx_pointer: *mut std::ffi::c_void,
+) {
+    if envelope.is_null() || tx_pointer.is_null() {
+        return;
+    }
+
+    let mut len = 0;
+    let buf = native::sentry_envelope_serialize(envelope, &mut len);
+
+    if !buf.is_null() && len > 0 {
+        let transport: Transport = std::mem::transmute(tx_pointer);
+        transport(std::slice::from_raw_parts(buf as *const u8, len as usize));
+    }
+
+    native::sentry_free(buf as _);
+}
+
 /// Captures process crashes and reports them to Sentry.
 ///
-/// Internally, this uses the Breakpad client to capture crash signals and send minidumps to Sentry.
-/// If no DSN is configured, the crash handler is not initialized.
-#[derive(Debug, Default)]
+/// Internally, this uses the Breakpad client to capture crash signals and create minidumps. If no
+/// DSN is configured, the crash handler is not initialized.
+///
+/// To send crashes to Sentry, configure a [`transport` function](Self::transport). Otherwise, the
+/// crash reporter writes crashes to a local database folder, where they can be handled manually.
+#[derive(Default)]
 #[cfg_attr(not(unix), allow(dead_code))]
 pub struct CrashHandler<'a> {
     dsn: &'a str,
     database: &'a str,
+    transport: Option<Transport>,
     release: Option<&'a str>,
     environment: Option<&'a str>,
 }
 
+impl<'a> fmt::Debug for CrashHandler<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let transport = match self.transport {
+            Some(_) => "Some(fn)",
```

Comment:
`fn(&[u8])` does not implement `Debug`, hence we have to do it ourselves. We could also wrap the transport in a custom type and implement `Debug` just for that, but I found this to be less code.